client-v2/src/shared/components/Collection.js

```diff
@@ -0,0 +1,74 @@
+// @flow
+
+import React from 'react';
+import { connect } from 'react-redux';
+import styled from 'styled-components';
+
+import type { Collection } from '../types/Collection';
+import type { State } from '../types/State';
+import Article from './Article';
+import GroupDisplay from './GroupDisplay';
+import {
+  createCollectionSelector,
+  selectSharedState
+} from '../selectors/shared';
+
+type ContainerProps = {
+  id: string, // eslint-disable-line react/no-unused-prop-types
+  selectSharedState: (state: any) => State // eslint-disable-line react/no-unused-prop-types
+};
+
+type Props = ContainerProps & {
+  collection: Collection,
+  stage: string
+};
+
+const CollectionContainer = styled('div')`
+  background-color: white;
+  padding: 5px;
+  margin: 5px;
+  color: black;
+`;
+
+const CollectionHeadline = styled('div')`
+  font-weight: bold;
+  padding: 7px;
+`;
+
+const collectionDetail = ({ collection, stage }: Props) =>
+  collection ? (
+    <CollectionContainer>
+      <CollectionHeadline>{collection.displayName}</CollectionHeadline>
+      {collection.groups
+        ? collection.groups
+            .slice()
+            .reverse()
```

Comment:
Is there any reason this doesn't come reversed from the selector (there may will be!)

Reply:
Nope, we could definitely do this earlier. Arguably at the point at which we insert the model into the state.